functorch/_src/eager_transforms.py

```diff
@@ -745,6 +753,15 @@ def jvp(func, primals, tangents, *, strict=False):
                                for p, t in zip(flat_primals, flat_tangents))
             duals = tree_unflatten(flat_duals, primals_spec)
             result_duals = func(*duals)
+            if has_aux:
+                if not (isinstance(result_duals, tuple) and len(result_duals) == 2):
+                    raise RuntimeError(
+                        f"{jvp_str}: output of function f should be a tuple: (output, aux) "
+                        "if has_aux is True"
+                    )
+                result_duals, aux = result_duals
+                aux = _undo_create_differentiable(aux, level)
```

Comment:
_undo_create_differentiable only works on Tensors right? Does jax permit returning arbitrary aux output? (If so, we don't necessarily have to fix that here, I'm just trying to understand)

Reply:
I made another PR where trying to remove restrictions on aux. Here I'm just keeping similar approach as other ops doing